WebhookSender.cs

```diff
@@ -0,0 +1,49 @@
+using System;
+using System.Net.Http;
+using System.Net;
+using System.Threading.Tasks;
+
+namespace SampleProject.Webhooks
+{
+    public class WebhookSender
+    {
+        private readonly HttpClient _httpClient;
+        private readonly IReportingService _reportingService;
+
+        public WebhookSender(HttpClient httpClient, IReportingService reportingService)
+        {
+            _httpClient = httpClient;
+        }
+
+        public async Task SendWebhook(WebhookConfiguration configuration, bool webhookIsActive)
+        {
+            var d = new WebhookDataStore("https://webhooks.checkout.com/datastore");
+            var additionalWebhookInfo = d.GetAdditionalInfo(configuration.WebhookId);
+
+            var httpRequest = new HttpRequestMessage(HttpMethod.Post, configuration.Url);
+            var webhookResponse = _httpClient.Send(httpRequest);
+
+            if (webhookResponse.StatusCode == HttpStatusCode.OK)
+            {
+                await d.UpdateResult("Success");
+                configuration.Success = true;
+            }
+            else if (webhookResponse.StatusCode == HttpStatusCode.NotFound)
+            {
+                await d.UpdateResult("Failed");
+            }
+            else if (webhookResponse.StatusCode == HttpStatusCode.ServiceUnavailable)
+            {
+                await d.UpdateResult("Failed");
+            }
+
+            try
```

Comment:
If d.UpdateResult can fail, please wrap the above calls to this method in the try block.

Exception handling requires some attention, if we are using exception handling per class (rather than ActionFilters etc), we should take appropriate action, e.g. retry/logging